lib/stripe/subscriptions/usage.ex

```diff
@@ -0,0 +1,106 @@
+defmodule Stripe.SubscriptionItem.Usage do
+  @moduledoc """
+  Work with Stripe usage record objects.
+
+  Stripe API reference: https://stripe.com/docs/api/usage_records
+  """
+
+  use Stripe.Entity
+  import Stripe.Request
+
+  @type t :: %__MODULE__{
+          id: Stripe.id(),
+          object: String.t(),
+          invoice: Stripe.id() | nil,
+          livemode: boolean,
+          quantity: non_neg_integer,
+          subscription_item: Stripe.id() | Stripe.SubscriptionItem.t(),
+          timestamp: Stripe.timestamp()
+        }
+
+  defstruct [
+    :id,
+    :object,
+    :invoice,
+    :livemode,
+    :quantity,
+    :subscription_item,
+    :timestamp
+  ]
+
+  @plural_endpoint "subscription_items"
+
+  @doc """
+  Creates a usage record for a specified subscription item id and date, and fills it with a quantity.
+  """
+  def create(id, params, opts \\ [])
+
+  @doc """
+  Creates a usage record for a specified subscription item id and date, and fills it with a quantity.
+  """
+  @spec create(Stripe.SubscriptionItem.t(), params, Stripe.options()) :: {:ok, t} | {:error, Stripe.Error.t()}
+        when params: %{
+              :quantity => float,
+              :timestamp => Stripe.timestamp() | non_neg_integer,
+              optional(:action) => String.t()
+             } 
+  def create(item, params, opts) when is_map(item) do
+    new_request(opts)
+    |> put_endpoint(build_url(item))
+    |> put_method(:post)
+    |> put_params(params)
```

Comment:
Here we need `|> put_param(:subscription_item, id)` since that is a required param to pass to stripe.

Reply:
Also, I think we can get rid of the first block here and just require an `id` to be passed instead of an object.

Reply:
I think that is a good call out. If you have the object, you can grab the id out of it before calling the function. I was just thinking you could streamline the process by allowing the object passed in.

Reply:
It is a good idea, but to keep parity with most other methods, we probably want to just pass the `id`.  THanks for sticking with this PR!